File: infrastructure/postgres-db/src/main/kotlin/com/donperry/persistence/pet/mapper/PetMapper.kt
package com.donperry.persistence.pet.mapper

import com.donperry.model.pet.Pet
import com.donperry.model.pet.Species
import com.donperry.persistence.pet.entities.PetData
import java.util.*

object PetMapper {
    fun toEntity(petModel: Pet): PetData =
        PetData(
            id = petModel.id?.let { UUID.fromString(it) },
            name = petModel.name,
            species = petModel.species.name,
            breed = petModel.breed,
            age = petModel.age,
            birthdate = petModel.birthdate,
            weight = petModel.weight,
            nickname = petModel.nickname,
            owner = petModel.owner,
            registrationDate = petModel.registrationDate,
            photoUrl = petModel.photoUrl
        )

    fun toModel(petData: PetData): Pet =
        Pet(
            id = petData.id.toString(),
            name = petData.name,
            species = Species.valueOf(petData.species),
            breed = petData.breed,
            age = petData.age,
            birthdate = petData.birthdate,
            weight = petData.weight,
            nickname = petData.nickname,
            owner = petData.owner,
            registrationDate = petData.registrationDate,
            photoUrl = petData.photoUrl
        )
}

File: front/src/features/pets/infrastructure/pet-mapper.ts
import type { Pet, PetSummary, Species } from '../domain/entities'

export interface PetDto {
  id: string
  name: string
  species: string
  breed: string | null
  age: number
  birthdate: string | null
  weight: number | null
  nickname: string | null
  owner: string
  registrationDate: string
  photoUrl: string | null
}

export interface PetListDto {
  id: string
  name: string
  species: string
  breed: string | null
  photoUrl: string | null
}

export function toPet(dto: PetDto): Pet {
  return {
    id: dto.id,
    name: dto.name,
    species: dto.species as Species,
    breed: dto.breed,
    age: dto.age,
    birthdate: dto.birthdate,
    weight: dto.weight,
    nickname: dto.nickname,
    owner: dto.owner,
    registrationDate: dto.registrationDate,
    photoUrl: dto.photoUrl,
  }
}

export function toPetSummary(dto: PetListDto): PetSummary {
  return {
    id: dto.id,
    name: dto.name,
    species: dto.species as Species,
    breed: dto.breed,
    photoUrl: dto.photoUrl,
  }
}
